src/bin/rdcore/stream_hash.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright 2020 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use error_chain::bail;
+use openssl::sha;
+use regex::Regex;
+use std::fs::OpenOptions;
+use std::io::{self, stdin, stdout, BufRead, BufReader, Read, Write};
+
+use crate::cmdline::*;
+use crate::errors::*;
+
+pub fn stream_hash(config: &StreamHashConfig) -> Result<()> {
+    let mut hash_file = OpenOptions::new()
+        .read(true)
+        .open(&config.hash_file)
+        .chain_err(|| format!("opening {}", config.hash_file))?;
+    do_stream_hash(&mut hash_file, &mut stdin(), &mut stdout())
+}
+
+fn do_stream_hash(
+    hash_file: &mut impl Read,
+    input: &mut impl Read,
+    output: &mut impl Write,
+) -> Result<()> {
+    // get buffered reader
+    let mut hash_file = BufReader::new(hash_file);
+
+    // read header line
+    let mut line = String::new();
+    if hash_file
+        .read_line(&mut line)
+        .chain_err(|| "reading hash file")?
+        == 0
+    {
+        bail!("hash file is empty");
+    }
+
+    // parse it
+    let captures = Regex::new(r"^stream-hash ([a-z0-9]+) ([0-9]+)\n$")
```

Comment:
Heh, same feeling about regex here as [there](https://github.com/coreos/coreos-installer/pull/153#discussion_r451780476). Feels like a `splitn` would be cleaner and possibly less lines, but good with this as well!